### PR TITLE
MGDCTRS-853:fix:forms don't retain state when going back in flow

### PR DESCRIPTION
**Pull request by uidoyen**

### CODE REVIEW

**Review by indraraj — APPROVED** [review]
LGTM :+1:

File: src/app/machines/CreateConnectorWizard.machine.ts [hunk 1]
```diff
@@ -334,7 +340,9 @@ export const creationWizardMachine = model.createMachine(
                     ? context.connectorTypeDetails
                     : context.selectedConnector,
                   configuration: context.duplicateMode
```
Comment (gashcrumb) [inline src/app/machines/CreateConnectorWizard.machine.ts:342]: I wonder, is `context.duplicateMode && typeof context.connectorConfiguration !== `undefined` ? context.connectorConfiguration : context.connectorData?.connector` the same, and do you find this style easy to follow?  Personally I tend to find nested conditionals really tricky to understand at least for a quick glance.  I'd advocate for combining the conditions if it yields the same result.

Normally I like to pull this kind of calculation out and use a variable, but it's not an option here.

Comment (uidoyen) [reply src/app/machines/CreateConnectorWizard.machine.ts:342]: @gashcrumb You are right and thanks for pointing this out! Nested condition is always tricky to understand. Actually, we should have a different context and machines for Duplicate connector flow. Probably, We would consider changing this in the near future which can solve this. :smile:

Comment (gashcrumb) [reply src/app/machines/CreateConnectorWizard.machine.ts:342]: hmm, yeah a separate machine for the duplicate flow would make sense, it's only a subtle difference from the regular create flow, but it's different enough.  I've been looking at user tracking, so I'll keep this in mind as I work on that effort.  Thanks!

**Review by gashcrumb — APPROVED** [review]

**Pull request merged**